icefoganalytics/travel-authorization
# Part 2: Departmental Administration: Create a Travel Auth for Another Employee Within My Department

Fixes https://github.com/icefoganalytics/travel-authorization/issues/166

Relates to:
- https://github.com/icefoganalytics/travel-authorization/issues/62

# Context

Give department admins the ability to create travel authorizations for another employee in there department.

## Requirements

- utilize the same UI as the /my-travel-requests, but will require the ability to pick the user the travel request is applied to.
- create travel request for other user button could be located in the /manage-travel-requests page or the /travel-request pages. /travel-request should be renamed to /travel-requests
- Route for edit access should be /travel-requests/46/details/edit
- Rename DEPT_ADMIN = "dept_admin", to `department_admin` for better future readability.

# Implementation

Build out edit details page for management of users.

# Screenshots

http://localhost:8080/travel-requests/81/details/edit
![image](https://github.com/icefoganalytics/travel-authorization/assets/23045206/5a49c678-020f-4bbf-b30c-24e62199dcda)

# Testing Instructions

1. Run the test suite via `dev test` (or `dev test_api`)
2. Boot the app via `dev up`
3. Log in to the app at http://localhost:8080
4. Go to the "Manager View" via the top drop down nav (http://localhost:8080/manage-travel-requests)
5. Create a new travel request via the "create travel request" button.
6. Check that all the fields in the edit "Details" page (http://localhost:8080/travel-requests/81/details/edit) works and is fully reactive.
> Note that the the "Estimate" and "Expense" pages have not been built.
